### DISCUSSION TITLE
vips copy of large tiff with transparency to png fails :-(

### DISCUSSION THREAD

**Discussion by dfrumin2** [libvips/libvips#3640] [Q&A]
Hi John - 

I have a very large TIF with transparency that I want to convert to a PNG.  Unfortunately after ~1sec I get an 8,246 PNG file that is invalid.  

This is only with the original TIF. when I want to keep transparency.  If I vips jpegsave it then it works great.  I can flatten it to another TIF and then to PNG (no problem).  But if I want to keep the transparency and go to PNG it  bonks :-(

Any ideas?

thanks!!!

**Comment by jcupitt** [comment]
Hi @dfrumin2,

It should work. Can you share the file that fails somehow?

**Comment by dfrumin2** [comment]
Hi @jcupitt - forgive the newbie question but how do I DM you a link in GitHub?  I've looked everywhere and can't find it :-)   Should I send it to your gmail?  *BIG* thanks.

**Reply by jcupitt** [reply]
Sure, send it to me on gmail.

**Comment by dfrumin2** [comment]
Sent.  thanks!!!

**Comment by jcupitt** [comment]
I got the file, thanks. 

Your TIFF is CMYKA, and there was a bug in 8.14.2 which stopped conversions from CMYKA working. It seems fine in 8.14.4.

Try updating and see if that fixes it.

https://github.com/libvips/libvips/releases/tag/v8.14.4

**Comment by dfrumin2** [comment]
I'm using vips-8.14.4 :-(   
I downloaded vips-dev-w64-all-8.14.4.  Any chance the bug fix didn't get to that one?

Thanks for looking at it!

dan

**Reply by jcupitt** [reply]
It should be there. I see:

```
$ vips copy 'WIN FY24C2-PRES MISXPillarTower-ProsumerHP-Alt 1-Room-overlay-el-gr.tif' x.png
$ vipsheader x.png
x.png: 28458x40176 ushort, 4 bands, rgb16, pngload
$ vips --vips-version
libvips 8.14.4
```

What do you get for those commands?

**Comment by dfrumin2** [comment]
No luck.  Maybe the windows binary is missing something?  See below...  Note how the PNG is ~8K

>>vips --vips-version
libvips 8.14.4

>>vips copy "WIN FY24C2-PRES MISXPillarTower-ProsumerHP-Alt 1-Room-overlay-el-gr.tif" test.png

>>vipsheader test.png
vipsheader: pngload: end of stream

>>dir
09/02/2023  02:31 PM             8,246 test.png
09/01/2023  02:11 AM        54,846,717 WIN FY24C2-PRES MISXPillarTower-ProsumerHP-Alt 1-Room-baselayer.pdf
09/01/2023  02:11 AM       202,307,394 WIN FY24C2-PRES MISXPillarTower-ProsumerHP-Alt 1-Room-el-gr.pdf
09/01/2023  02:11 AM       101,730,256 WIN FY24C2-PRES MISXPillarTower-ProsumerHP-Alt 1-Room-overlay-el-gr.tif

**Reply by jcupitt** [reply]
Ah! I think I've found a bug, hang on.

**Reply by jcupitt** [reply]
I think I fixed the bug: https://github.com/libvips/libvips/pull/3647

**Comment by dfrumin2** [comment]
Awesome!  Do you know when a binary will be available that I can test/use?

One more question - we're also using vips in a linux docker container for the PDF => JPG/PNG export.  Unfortunately we're seeing the output come out in garbled colors and goobered pixels.  Do you think it's not using the default ICC/ICM that's embedded?  My guys are looking to see if maybe the container is missing lcms2 or something.  We use Windows to design our pipeline and then we run it in linux docker container for scale...

Thanks again!!!

![image](https://github.com/libvips/libvips/assets/114293525/c24b5dc3-cce1-400b-9049-3d1ae31ed263)

![image](https://github.com/libvips/libvips/assets/114293525/2a6a21e5-f46b-45ad-898d-ac8e17392c62)

**Reply by jcupitt** [reply]
You can build it yourself, of course, otherwise a few months I suppose.

I don't know what's causing the colour error, you'd need to share a test case.

**Comment by dfrumin2** [comment]
Thanks John - I'll get a test file for you of the color change.

Question on the CMYKA bug you found - is there a workaround or other thing we can do with libvips when we're generating that TIFF?  For various compliance reasons we try to stay to official releases of libraries so hoping there's something we can do with 8.14.4.  And since we generate it maybe there's a flag or a way to do it that avoids the downstream issue?  I ask because so far this is the only TIFF we found that fails - the others we generate with libvips all work great.  :-)

THANKS!

**Reply by jcupitt** [reply]
The bug was with automatic handling of alpha during `icc_import` and `icc_export`. The workarounds would be to either handle alpha yourself, or to use `icc_transform`. The details would depend on how you're calling libvips. Is this via Python, for example? Or on node?

You don't need to stay on releases -- just cherrypick that commit into your 8.14.4. If you build your own libvips in docker it's just as reproducible and auditable as a proper release.

**Comment by dfrumin2** [comment]
We're doing it using code but I'll ask Fabian on our eng team to comment.  RE: releases, I'll ask him to look at that too.  Compliance is a big thing in our universe with our clients so trying to stay on "official" releases has a benefit in less explaining :-)
